src/libraries/System.Text.Json/gen/JsonSourceGenerator.DiagnosticDescriptors.cs

```diff
@@ -72,6 +72,14 @@ internal static class DiagnosticDescriptors
                 category: JsonConstants.SystemTextJsonSourceGenerationName,
                 defaultSeverity: DiagnosticSeverity.Warning,
                 isEnabledByDefault: true);
+
+            public static DiagnosticDescriptor JsonStringEnumConverterNotSupportedInAot { get; } = new DiagnosticDescriptor(
```

Comment:
Do we need this new diagnostic? Isn't the `RequiresDynamicCode` attribute on `JsonStringEnumConverter` enough to devs that `JsonStringEnumConverter` is not supported in native AOT?

Reply:
It was added in reaction to https://github.com/dotnet/runtime/pull/87149#discussion_r1219302123 not triggering at all either at design time or at compile time. If we can get to the point that this always warns then I could remove that diagnostic.

Reply:
That comment will not be a problem in publish - both trimmer and AOT compiler will correctly warn if necessary.
The only "hole" is in the analyzer - which will not report this warning currently.
That said - if the specific diagnostic will always provide better UX than the more general RDC warning (even if the analyzer did report it) - so it might be worth keeping this just for the better UX.

Downside is that if we fix the analyzer this might produce two warnings.

Reply:
Can we deprecate the diagnostic in the future if the analyzer does get fixed?

Reply:
Probably - we may need to figure out some versioning stuff, but it shouldn't be too hard.